lib/puppet-lint/plugins/check_classes.rb

```diff
@@ -100,42 +99,46 @@ def check
 # https://docs.puppet.com/guides/style_guide.html#display-order-of-parameters
 PuppetLint.new_check(:parameter_order) do
   def check
-    (class_indexes + defined_type_indexes).each do |class_idx|
-      unless class_idx[:param_tokens].nil?
-        paren_stack = []
-        hash_or_array_stack = []
-        class_idx[:param_tokens].each_with_index do |token, i|
-          if token.type == :LPAREN
-            paren_stack.push(true)
-          elsif token.type == :RPAREN
-            paren_stack.pop
-          elsif token.type == :LBRACE || token.type == :LBRACK
-            hash_or_array_stack.push(true)
-          elsif token.type == :RBRACE || token.type == :RBRACK
-            hash_or_array_stack.pop
-          end
-          next if (! hash_or_array_stack.empty?)
-          next unless paren_stack.empty?
+    (class_indexes + defined_type_indexes).reject { |class_idx|
+      class_idx[:param_tokens].nil?
+    }.each do |class_idx|
```

Comment:
Rather than iterate through the array twice here, I would suggest keeping the original each iterator and using `next if class_idx[:param_tokens].nil?` at the start of the block.